pkg/metrics/cluster/config_watcher.go

```diff
@@ -131,6 +144,12 @@ func (w *configWatcher) Refresh(ctx context.Context) (err error) {
 		reshardDuration.WithLabelValues(success).Observe(time.Since(start).Seconds())
 	}()
 
+	// This is used to determine if the context was already exceeded before calling the kv provider
+	if err = ctx.Err(); err != nil {
+		return fmt.Errorf("context deadline exceeded before calling store.all")
```

Comment:
nit: I'm a little uncomfortable with this since it breaks common expectations about what a timeout error looks like in Go, plus the deadline can still expire in between this check and invoking w.store.All.

Reply:
Reformatted this